play.pokemonshowdown.com/src/client-connection.ts

```diff
@@ -166,9 +168,10 @@ export class PSConnection {
 	private retryConnection() {
 		if (!this.canReconnect()) return;
 		this.reconnectTimer = setTimeout(() => {
+			clearTimeout(this.reconnectTimer!);
 			this.reconnectTimer = null;
 			if (!this.connected && this.canReconnect()) {
-				PS.send('/reconnect');
+				PS.room.send('/reconnect');
```

Comment:
```suggestion
				PS.mainmenu.send('/reconnect');
```

`PS.room` isn't always capable of sending.